CMakeLists.txt

```diff
@@ -202,19 +218,58 @@ endif (ENABLE_CLANG_AS_LIB)
 ##             TEST APP                      ##
 ###############################################
 
+## TestJit.cpp
+#----- Sources
+
+# It does make no sense building it if there is no LLVM, given that the JIT implementation relies on LLVM. 
+if (ENABLE_CLANG_AS_LIB)
+  set (VC_TESTJIT_APP_SRC TestJit.cpp)
+
+  set (VC_EXE_NAME libVC_testJit)
+
+  if ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    set_source_files_properties(${VC_TESTJIT_APP_SRC} PROPERTIES
+              COMPILE_FLAGS "-DVC_DEBUG -DHAVE_CLANG_AS_LIB -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
+  )
+  else( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    set_source_files_properties(${VC_TESTJIT_APP_SRC} PROPERTIES
+              COMPILE_FLAGS "-DHAVE_CLANG_AS_LIB -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
+    )
+  endif( CMAKE_BUILD_TYPE MATCHES DEBUG )
+
+
+  add_executable(${VC_EXE_NAME} ${VC_TESTJIT_APP_SRC})
+
+  link_directories(${LLVM_LIBRARY_DIR} ${LIBCLANG_LIBRARY_DIR} "${CMAKE_INSTALL_PREFIX}/lib")
+  target_link_libraries(${VC_EXE_NAME} ${VC_LIB_NAME} ${VC_LIB_DEPS})
+endif (ENABLE_CLANG_AS_LIB)
+
+## Test.cpp
 #----- Sources
 set ( VC_TEST_APP_SRC Test.cpp )
 
 set ( VC_EXE_NAME libVC_test )
 
 if (ENABLE_CLANG_AS_LIB)
-  set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
-                              COMPILE_FLAGS "-flto -DHAVE_CLANG_AS_LIB -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
-                              )
+  if ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
+                                COMPILE_FLAGS "-DVC_DEBUG -DHAVE_CLANG_AS_LIB -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
+                                )
+  else ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
+                                COMPILE_FLAGS "-DHAVE_CLANG_AS_LIB -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
+                                )
+  endif ( CMAKE_BUILD_TYPE MATCHES DEBUG )
 else (ENABLE_CLANG_AS_LIB)
-  set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
-                              COMPILE_FLAGS "-flto -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
-                              )
+  if ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
+                                COMPILE_FLAGS "-DVC_DEBUG -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
+                                )
+  else ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
+                                COMPILE_FLAGS "-DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
+                                )
+  endif ( CMAKE_BUILD_TYPE MATCHES DEBUG )
```

Comment:
possibly this can be rewritten with `target_compile_definitions` or `COMPILE_DEFINITIONS` property.